optuna/integration/allennlp.py

```diff
@@ -112,6 +286,37 @@ def __init__(
         else:
             self._include_package = include_package
 
+        storage = trial.study._storage
+
+        if isinstance(storage, optuna.storages.RDBStorage):
+            url = storage.url
+        elif isinstance(storage, optuna.storages.RedisStorage):
+            url = storage._url
+        elif isinstance(storage, optuna.storages._CachedStorage):
+            assert isinstance(storage._backend, optuna.storages.RDBStorage)
+            url = storage._backend.url
+        else:
+            url = ""
```

Comment:
When I use the in-memory storage, I got the following error due to this line:
```console
[W 2020-09-16 15:24:13,416] Trial 0 failed because of the following error: ArgumentError("Could not parse rfc1738 URL from string ''")
```
I think it is not easy for users to find the cause since they do not provide the storage URL by themselves. Is it possible to tell users to use the `RDBStorage` or `RedisStorage` when they use in-memory storage?

Reply:
Thank you for the catch.
I had to check whether `storage` is an empty string or not.
In d3f8837, I update the PR for it.

Now the following message appears when users attempt to use in-memory storage.

```
RuntimeError: If you want to use AllenNLPExecutor and AllenNLPPruningCallback, you have to use RDB or Redis storage.
```

Reply:
Thank you for your update. I confirmed the error message.
SGTM although the error handling is getting a bit complicated by https://github.com/optuna/optuna/commit/d3f8837bd69f76410433525e42133ae888e5e7f2.